keras/preprocessing/sequence.py

```diff
@@ -216,3 +218,93 @@ def _remove_long_seq(maxlen, seq, label):
+
+    # Arguments
+        data: indexable generator (such as list) containing
+            consecutive data points.
+        targets: targets corresponding to instances in `data`.
+            Should have same length as data.
+        lookback: length of history to consider for prediction.
+            data[i-lookback : i+1] is used as input.
+        delay: how further in the future we are looking to predict.
+            targets[i + delay] becomes the target for
+            data[i-lookback : i+1]
+        sampling_rate: period between successive individual data
+            points. For rate `r`, points
+            data[i], data[i-r], ... data[i - lookback] are used as input.
+        stride: period between successive sub-sequences used for
+            training. For stride s, consecutive training samples would
+            be centered around data[i], data[i+s], data[i+2*s], ...
+        start_index, end_index: delimiters for sub-sets of data & targets
+            to be used for batch generation. Useful to reserve part of
+            data for validation.
+        shuffle: whether to shuffle training samples, or draw them in
+            chronological order.
+        reverse: if true, input points are provided in reverse order.
+        batch_size: size of successful batches (except maybe last one)
+    
+    # Returns
+        A `Sequence` instance.
+    """
+    def __init__(self, data, targets,
+                 lookback,
+                 delay,
+                 sampling_rate=1,
+                 stride=1,
+                 start_index=0,
+                 end_index=None,
+                 shuffle=False,
+                 reverse=False,
+                 batch_size=128):
+        self.data = data
+        self.targets = targets
+        if not hasattr(self.targets[0], "__len__"):
+            self.output_dim = 1
+        else:
+            self.output_dim = len(self.targets[0])
+        self.lookback = lookback
+        self.delay = delay
+        self.sampling_rate = sampling_rate
+        self.stride = stride
+        self.start_index = start_index + lookback
+        if end_index == None or end_index + delay > len(data) - 1:
+            end_index = len(data) - delay - 1
+        self.end_index = end_index
+        self.shuffle = shuffle
+        self.reverse = reverse
+        self.batch_size = batch_size
+
+    def __len__(self):
+        return np.ceil(
+                (self.end_index - self.start_index) / 
+                (self.batch_size * self.stride))
+
+    def __getitem__(self, index):
+        if self.shuffle:
+            rows = np.random.randint(
+                self.start_index, self.end_index, size=self.batch_size)
+        else:
+            i = self.start_index + self.batch_size * self.stride * index
+            rows = np.arange(i, min(i + self.batch_size * self.stride, self.end_index), 
+                    self.stride)
+
+        samples = np.zeros((len(rows),
```

Comment:
pretty sure you can use `np.empty` here instead of np.zeros

Reply:
Ack.